touch/src/main.rs

```diff
@@ -0,0 +1,240 @@
+use chrono::NaiveDateTime;
+use clap::{load_yaml, App, ArgMatches};
+use filetime::{set_file_atime, set_file_mtime, FileTime};
+use std::fs::{metadata, File};
+use std::io::Result;
+use std::process;
+use std::time::SystemTime;
+
+fn main() {
+    let yaml = load_yaml!("touch.yml");
+    let matches = App::from_yaml(yaml).get_matches();
+
+    // get files list in argument
+    let files = if matches.is_present("FILE") {
+        matches.values_of("FILE").unwrap().collect::<Vec<_>>()
+    } else {
+        Vec::new()
+    };
+
+    if files.len() > 0 {
+        touch(&matches, &files).unwrap();
+    } else {
+        eprintln!("touch: a file name is required.\n");
+        process::exit(1);
+    }
+}
+
+fn touch(matches: &ArgMatches, files: &Vec<&str>) -> Result<()> {
+    for filename in files {
+        // if file already exist in the current directory
+        let file_metadata = metadata(&filename);
+        let no_create = !matches.is_present("nocreate");
+        if file_metadata.is_err() && no_create {
+            match File::create(&filename) {
+                Ok(_) => (),
+                Err(e) => eprintln!("touch: Failed to create file {} : {}", &filename, e),
+            }
+        } else {
+            let date = matches.is_present("date");
+            if date {
+                let date_value = matches.value_of("date").unwrap();
+                let native_date =
+                    NaiveDateTime::parse_from_str(&date_value, "%Y-%m-%d %H:%M:%S").unwrap();
```

Comment:
I think it's here that are causing the panic, handle that and print the parsing error to stderr